lang/cs/Org.Apache.REEF.IMRU.Tests/JobLifecycleManagerTest.cs

```diff
@@ -0,0 +1,210 @@
+        private IEnumerable<TestObserver> JobLifeCycleMangerEventTest(
+            IJobCancelledDetector detector,
+            bool subscribeObserver = true,
+            int observerCount = 1,
+            int signalCheckPeriodSec = 1,
+            int waitForEventPeriodSec = 2)
+        {
+            var manager = Activator.CreateInstance(
+                typeof(JobLifeCycleManager),
+                BindingFlags.NonPublic | BindingFlags.Instance,
+                null,
+                new object[] { detector, signalCheckPeriodSec },
+                null,
+                null) as JobLifeCycleManager;
+
+            var observers = Enumerable.Range(1, observerCount)
+                .Select(_ => subscribeObserver ? new TestObserver(manager) : new TestObserver(null))
+                .ToList();
+
+            manager.OnNext(NewStartedEvent());
+
+            Thread.Sleep(waitForEventPeriodSec * 1000);
+
+            return observers;
+        }
+
+        private void AssertCancelEvent(TestObserver observer, bool expectedEvent, string expectedMessage = null)
+        {
+            if (expectedEvent)
+            {
+                Assert.NotNull(observer.LastEvent);
+                Assert.Same(expectedMessage, observer.LastEvent.Message);
+            }
+            else
+            {
+                Assert.Null(observer.LastEvent);
+            }
+        }
+
+        private IDriverStarted NewStartedEvent()
+        {
+            // event is not really used by the driver, so can use null here
+            return null;
+        }
+
+        /// <summary>
+        /// Test helper class to provide predefined cancel signal for testing
+        /// </summary>
+        private class DetectorStubStatic : IJobCancelledDetector 
+        {
+            private bool IsCancelledResponse { get; set; }
+            private string CancellationMessage { get; set; }
+            private Action ActionOnIsCancelledCall { get; set; }
+
+            public DetectorStubStatic(bool isCancelledResponse, string expectedMessage = null, Action testAction = null)
+            {
+                IsCancelledResponse = isCancelledResponse;
+                CancellationMessage = expectedMessage;
+                ActionOnIsCancelledCall = testAction;
+            }
+
+            public bool IsJobCancelled(out string cancellationMessage)
+            {
+                if (ActionOnIsCancelledCall != null)
+                {
+                    ActionOnIsCancelledCall();
+                }
+
+                cancellationMessage = CancellationMessage;
+                return IsCancelledResponse;
+            }
+        }
+
+        /// <summary>
+        /// Test helper class to record JobCancelled events from lifecycle manager
+        /// </summary>
+        private class TestObserver : IObserver<IJobCancelled> 
+        {
+            public IJobCancelled LastEvent { get; private set; }
+
+            public TestObserver(IObservable<IJobCancelled> eventSource)
```

Comment:
fixed

---
In reply to: [86007642](https://github.com/apache/reef/pull/1161#discussion_r86007642) [](ancestors = 86007642)